service_container/injection_types.rst

```diff
@@ -248,3 +248,121 @@ to setter injection but with these additional important problems:
 But, it is useful to know that this can be done with the service container,
 especially if you are working with code that is out of your control, such
 as in a third party library, which uses public properties for its dependencies.
+
+Static Injection
+----------------
+
+Another possible injection in to use a method which return `static`, 
+this approach allow you to make a service immutable::
+
+    // ...
+
+    class NewsletterManager
+    {
+        private $mailer;
+        
+        public function __construct(?MailerInterface $mailer = null)
+        {
+            $this->mailer = $mailer;
+        }
+        
+        /**
+         *
+         * The @return is required.
```

Comment:
I wouldn't add this to the code block. If it needs to be somewhere, that's in the doc itself.
Instead, we could replace it with `@required` + link to the doc page that tells what this does.